tracer/test/Datadog.Trace.Security.IntegrationTests/AspNetBase.cs

```diff
@@ -111,6 +115,25 @@ public async Task TestBlockedRequestAsync(MockTracerAgent agent, bool enableSecu
                     var attackEvent = item;
                     var shouldBlock = expectedStatusCode == HttpStatusCode.Forbidden;
                     Assert.Equal("Finds basic MongoDB SQL injection attempts", attackEvent.Rule.Name);
+
+                    // presences of json tag implies span should carry other security tags
+                    var securityTags = new Dictionary<string, string>()
+                    {
+                        { "appsec.event", "true" },
+                        { "_dd.origin", "appsec" },
+                        { "http.useragent", "Mistake Not..." },
+                        { "network.client.ip", "127.0.0.1" },
+                        { "actor.ip", "86.242.244.246" },
+                        { "http.request.headers.host", $"localhost:{_httpPort}" },
+                        { "http.request.headers.x-forwarded", "86.242.244.246" },
+                        { "http.request.headers.user-agent", "Mistake Not..." },
+                        { "http.response.headers.content-type", "text/plain; charset=utf-8" },
+                    };
+                    foreach (var kvp in securityTags)
+                    {
+                        Assert.True(span.Tags.TryGetValue(kvp.Key, out var tagValue), $"The tag {kvp.Key} was not found");
+                        Assert.Equal(kvp.Value, tagValue);
+                    }
```

Comment:
Is it worth snapshot testing this?

Reply:
Yes, we should do that, but I'll make that separate task.